src/moi_nlp_model.jl

```diff
@@ -413,3 +448,45 @@ function NLPModels.hprod!(
   end
   return hv
 end
+
+function NLPModels.jth_hprod!(
+  nlp::MathOptNLPModel,
+  x::AbstractVector,
+  v::AbstractVector,
+  j::Integer,
+  hv::AbstractVector,
+)
+  increment!(nlp, :neval_jhprod)
+  @rangecheck 1 nlp.meta.ncon j
+  hv .= 0.0
+  if nlp.meta.nlin + 1 ≤ j ≤ nlp.meta.nlin + nlp.quadcon.nquad
+    qcon = nlp.quadcon.constraints[j - nlp.meta.nlin]
+    coo_sym_add_mul!(qcon.A.rows, qcon.A.cols, qcon.A.vals, v, hv, 1.0)
+  end
+  if nlp.meta.nlin + nlp.quadcon.nquad + 1 ≤ j ≤ nlp.meta.ncon
+    nlp.λ[j - nlp.meta.nlin - nlp.quadcon.nquad] = 1.0
+    MOI.eval_hessian_lagrangian_product(nlp.eval, hv, x, v, 0.0, nlp.λ)
+    nlp.λ[j - nlp.meta.nlin - nlp.quadcon.nquad] = 0.0
+  end
+  return hv
+end
+
+function NLPModels.ghjvprod!(
+  nlp::MathOptNLPModel,
+  x::AbstractVector,
+  g::AbstractVector,
+  v::AbstractVector,
+  ghv::AbstractVector,
+)
+  increment!(nlp, :neval_hprod)
+  ghv .= 0.0
+  for i in nlp.meta.nlin + 1 : nlp.meta.nlin + nlp.quadcon.nquad
+    qcon = nlp.quadcon.constraints[i - nlp.meta.nlin]
+    ghv[i] = coo_sym_dot(qcon.A.rows, qcon.A.cols, qcon.A.vals, g, v)
+  end
+  for i in nlp.meta.nlin + nlp.quadcon.nquad + 1 : nlp.meta.ncon
+    jth_hprod!(nlp, x, v, i, nlp.hv)
```

Comment:
This is your extra evaluations
```suggestion
    jth_hprod!(nlp, x, v, i, nlp.hv)
    decrement!(nlp, :neval_jhprod)
```

Reply:
This is still the line you need to add for the counters